src/m365/spo/commands/file/file-rename.spec.ts

```diff
@@ -0,0 +1,312 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../../appInsights';
+import auth from '../../../../Auth';
+import { Cli, Logger } from '../../../../cli';
+import Command, { CommandError } from '../../../../Command';
+import request from '../../../../request';
+import { sinonUtil } from '../../../../utils';
+import commands from '../../commands';
+const command: Command = require('./file-rename');
+const fileRemoveCommand: Command = require('./file-remove');
+
+describe(commands.FILE_RENAME, () => {
+  let log: any[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+
+  const renameResponseJson = [
+    {
+      'ErrorCode': 0,
+      'ErrorMessage': null,
+      'FieldName': 'FileLeafRef',
+      'FieldValue': 'test 2.docx',
+      'HasException': false,
+      'ItemId': 642
+    }
+  ];
+
+  const renameValue = {
+    value: renameResponseJson
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    (command as any).items = [];
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get, 
+      request.post,
+      Cli.executeCommandWithOutput
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      appInsights.trackEvent,
+      auth.restoreAuth
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.FILE_RENAME), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if the webUrl option is not a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { webUrl: 'foo', sourceUrl: 'abc', targetUrl: 'abc' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the webUrl option is a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', sourceUrl: 'abc', targetUrl: 'abc' } });
+    assert.strictEqual(actual, true);
+  });
+
+  it('should command complete successfully', (done) => {
```

Comment:
Let's rename this test to more clearly depict what we're testing, like: `forcefully renames file from a non-root site in the root folder of a document library when a file with the same name exists (or it doesn't?)`